src/libraries/Microsoft.Extensions.Options.ConfigurationExtensions/tests/SourceGenerationTests/Microsoft.Extensions.Options.ConfigurationExtensions.SourceGeneration.Tests.csproj

```diff
@@ -2,9 +2,9 @@
   <PropertyGroup>
     <Nullable>enable</Nullable>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkMinimum)</TargetFrameworks>
-    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <DefineConstants>$(DefineConstants);BUILDING_SOURCE_GENERATOR_TESTS;ROSLYN4_0_OR_GREATER;ROSLYN4_4_OR_GREATER</DefineConstants>
-    <!-- The SDK disables the configuration binding generator by default no matter how it was referenced, so we need to enable it here for testing. -->
+    <Features>$(Features);InterceptorsPreview</Features>
```

Comment:
>InterceptorsPreview [](http://example.com/codeflow?start=26&length=19)

Is this required for options source gen? note that options source gen can run with Language Version 8 and above. I am not sure how InterceptorsPreview will affect that.

Reply:
If the binding generator is also enabled in the project, options generator users will get a compiler error saying that they need to bump the language version.

Reply:
What happens if the binding generator is not enabled? does the compiler will complain on options source gen? I am trying to understand how `InterceptorsPreview` work in general.

Reply:
>  I am trying to understand how InterceptorsPreview work in general. what happen if binding generator is enabled but no config binding calls found in the code?

Looping in @RikkiGibson to confirm but I believe the compiler will only issue a warning if it discovers `InterceptsLocation` attributes without the feature flag being set.

So if the binding generator is enabled but no config binding calls were found and code and as a result no interceptor methods were created it should be fine.

Reply:
I think using InterceptsLocationAttribute without enabling the feature is an error. It's fine to enable the feature but not use it.

Reply:
@RikkiGibson just to confirm, enabling the feature and not using it while using previous compiler versions (like v10 and below) will be fine and we'll not get any error or warning. right?

Reply:
That's right, compiler issues no diagnostic when an unknown feature name is used in `<Features>`.

Reply:
I consider this resolved given the discussion, so I'll proceed as-is.